docs/reference-guide/graph-algorithms.md

```diff
@@ -312,4 +313,79 @@ property equal to `false` and to nodes with a `drinks_USD` property less than `1
 ```cypher
 MATCH path=(n {id: 0})-[*WSHORTEST (r, n | n.total_USD) total_weight (r, n | r.eu_border = false AND n.drinks_USD < 15)]-(m {id: 46})
 RETURN path,total_weight;
+```
+
+## All Shortest Paths
+
+Finding all shortest paths is an expansion of the weighted shortest paths problem. The goal
+of finding the shortest path is obtaining any minimum sum of weights on the path from one
+node to the other. However, there could be multiple similar-weighted paths, and this algorithm
+enables to fetch them all.
+
+Weighted shortest path implementation returns only one-row resulting path from one
+node to the other. Commonly, multiple shortest paths are flowing through different
+routes. Syntax of obtaining all shortest paths is similar to the weighted shortest path
+and boils down to calling `[*ALLSHORTEST (r, n | r.weight)]` where `r` and `n` define 
+the current expansion relationship and node respectively.
+
+### Getting various results
+
+We'll use a similar example as the one above, used for the weighted shortest path.
+To showcase the characteristics of all shortest paths, we'll use a default weight equal to 1 in the next example.
+
+```cypher
+MATCH path=(n {id: 0})-[:CloseTo *ALLSHORTEST (r, n | 1)]-(m {id: 15})
+RETURN path;
+```
+
+Query returns multiple results, all with the 4 hops meaning that there are 
+multiple paths flowing from source node to the destination node.
+
+Nothing prevents the user to run a weighted query based on the weight property on
+each visited edge:
+
+```cypher
+MATCH path=(n {id: 0})-[:Type *ALLSHORTEST (r, n | r.weight)]-(m {id: 5})
+RETURN path;
+```
+
+One might want to obtain all relationships on all shortest paths. To 
+unwind them all,
+a little more complicated query needs to be written:
+
+```cypher
+MATCH path=(n {id: 0})-[relationships:CloseTo *ALLSHORTEST (r, n | n.total_USD)]-(m {id: 51})
+UNWIND (relationships(path)) AS edge
+RETURN DISTINCT edge; 
+```
+
+Information about total weight can be gathered in a similar fashion as in 
+the weighted shortest path. 
+```cypher
+MATCH path=(n {id: 0})-[relationships:CloseTo *WSHORTEST (r, n | 1) total_weight]-(m {id: 9})
+RETURN nodes(path), total_weight;
+```
+
+### Constraining the path's length
+
+All shortest path allows for upper bound path restriction. This significantly modifies the outcome of the shortest path if the globally shortest path is obtained from some route with more hops than the set upper bound. Finding the all shortest paths with path restriction
```

Comment:
```suggestion
All Shortest Path allows restricting the upper bound path. This addition significantly modifies the outcome of the algorithm if the globally shortest path is obtained from a route with more hops than the set upper bound. Finding the all shortest paths with path restriction
```

This globally is strange here. It's an adverb so it should stand next to a verb, not a noun. Can we say global shortest path? or "... if the shortest path is obtained globally..."?

Reply:
"All Shortest Paths allows restricting the upper bound path". Seems like "upper bound path" is a term